lib/shoryuken/manager.rb

```diff
@@ -1,202 +1,122 @@
-require 'shoryuken/processor'
-require 'shoryuken/fetcher'
-
 module Shoryuken
   class Manager
-    include Celluloid
     include Util
 
-    attr_accessor :fetcher
-    attr_accessor :polling_strategy
-
-    exclusive :dispatch
-
-    trap_exit :processor_died
-
     BATCH_LIMIT = 10
 
-    def initialize(condvar)
+    def initialize(fetcher, polling_strategy)
       @count = Shoryuken.options[:concurrency] || 25
       raise(ArgumentError, "Concurrency value #{@count} is invalid, it needs to be a positive number") unless @count > 0
+
       @queues = Shoryuken.queues.dup.uniq
-      @finished = condvar
 
-      @done = false
+      @done = Concurrent::AtomicBoolean.new(false)
 
-      @busy_processors  = []
-      @busy_threads = {}
-      @ready_processors = @count.times.map { build_processor }
+      @fetcher = fetcher
+      @polling_strategy = polling_strategy
+
+      @heartbeat = Concurrent::TimerTask.new(run_now: true, execution_interval: 0.25, timeout_interval: 60) { dispatch }
```

Comment:
I know, it just hurts when I know that it is not that difficult to get it optimal :). I will try do implement what we discussed in other PR and then we can talk again about different tradeoffs we're making there. Does that sound right?